backend/src/data/migrations/20210817163051_update_episode_status_enum.ts

```diff
@@ -0,0 +1,43 @@
+import { Knex } from 'knex';
+
+const TABLE_NAME = 'episodes';
+const COLUMN_NAME = 'status';
+
+const EpisodeStatus = {
+  PUBLISHED: 'published',
+  STAGING: 'staging',
+  LIVE: 'live',
+} as const;
+
+const formatAlterTableEnumSql = (
+  TABLE_NAME: string,
+  COLUMN_NAME: string,
+  EpisodeStatus: string[],
+): string => {
+  const constraintName = `${TABLE_NAME}_${COLUMN_NAME}_check`;
+  return [
+    `ALTER TABLE ${TABLE_NAME} DROP CONSTRAINT IF EXISTS ${constraintName};`,
+    `ALTER TABLE ${TABLE_NAME} ADD CONSTRAINT ${constraintName} CHECK (${COLUMN_NAME} = ANY (ARRAY['${EpisodeStatus.join(
+      '\'::text, \'')}'::text]));`,
+  ].join('\n');
+};
+
+export async function up(knex: Knex): Promise<void> {
+  return await knex.raw(
+    formatAlterTableEnumSql(TABLE_NAME, COLUMN_NAME, [
+      EpisodeStatus.PUBLISHED,
+      EpisodeStatus.STAGING,
+      EpisodeStatus.LIVE,
+    ]),
+  );
+}
+
+export async function down(knex: Knex): Promise<void> {
+  return await knex.raw(
+    formatAlterTableEnumSql(TABLE_NAME, COLUMN_NAME, [
+      EpisodeStatus.PUBLISHED,
+      EpisodeStatus.STAGING,
+      EpisodeStatus.LIVE,
```

Comment:
but on down we shouldn't have the `live` type, right?

Reply:
Yes, this is so, there is simply a problem with rolling back the migration if we already have an episode with live status, it will be difficult for us to roll back this migration

Reply:
Or is it normal?

Reply:
Okay, in this case everything fine 👍